apx/src/qapx_client.cpp

```diff
@@ -65,11 +68,17 @@ void Client::close()
    }
 }
 
-void Client::inPortDataNotification(NodeData *nodeData, QApxSimplePort *port, QVariant &value)
+void Client::inPortDataNotification(NodeData *nodeData, QApxSimplePort *port, const QVariant &value)
 {
    (void) nodeData;
-   QString name(port->getName());
-   emit requirePortData(port->getPortIndex(), name, value);
+   if (mInPortNotifyWithName)
```

Comment:
Using two different signals is a temporary solution. Better to change this to a signal that gives a reference or pointer to the port itself. Then the user can take the name or index as they see fit.

Will change this later, after this has been merged.